Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishTimePeriodExtractorConfiguration.cs

```diff
@@ -1,80 +1,84 @@
+
+        public static readonly Regex PureNumBetweenAnd =
+            new Regex(
+                $@"(entre\s+(la(s)?\s+)?)({BaseTimeExtractor.HourRegex}|{HourNumRegex})(\s*(?<leftDesc>{SpanishTimeExtractorConfiguration.DescRegex}))?\s*y\s*(la(s)?\s+)?({BaseTimeExtractor.HourRegex}|{HourNumRegex})\s*(?<rightDesc>{SpanishTimeExtractorConfiguration.PmRegex}|{SpanishTimeExtractorConfiguration.AmRegex}|{SpanishTimeExtractorConfiguration.DescRegex})?",
+                RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+        public static readonly Regex UnitRegex =
+            new Regex(@"(?<unit>horas|hora|h|minutos|minuto|mins|min|segundos|segundo|secs|sec)\b",
+                RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+        public static readonly Regex FollowedUnit = new Regex($@"^\s*{UnitRegex}",
+            RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+        public static readonly Regex NumberCombinedWithUnit = 
+            new Regex($@"\b(?<num>\d+(\,\d*)?)\s*{UnitRegex}", 
+                RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+        private static readonly Regex FromRegex = new Regex(@"((desde|de)(\s*la(s)?)?)$", RegexOptions.IgnoreCase | RegexOptions.Singleline);
+        private static readonly Regex ConnectorAndRegex = new Regex(@"(y\s*(la(s)?)?)$", RegexOptions.IgnoreCase | RegexOptions.Singleline);
+        private static readonly Regex BeforeRegex = new Regex(@"(entre\s*(la(s)?)?)", RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+        public SpanishTimePeriodExtractorConfiguration()
+        {
+            SingleTimeExtractor = new BaseTimeExtractor(new SpanishTimeExtractorConfiguration());
+            UtilityConfiguration = new SpanishDatetimeUtilityConfiguration();
+        }
+        public IDateTimeUtilityConfiguration UtilityConfiguration { get; }
+
+        public IExtractor SingleTimeExtractor { get; }
+
+        public IEnumerable<Regex> SimpleCasesRegex => new Regex[] { PureNumFromTo, PureNumBetweenAnd };
+
+        Regex ITimePeriodExtractorConfiguration.TillRegex => SpanishDatePeriodExtractorConfiguration.TillRegex;
+
+        Regex ITimePeriodExtractorConfiguration.NightRegex => SpanishDateTimeExtractorConfiguration.NightRegex;
+
+        public bool GetFromTokenIndex(string text, out int index)
+        {
+            index = -1;
+            var fromMatch = FromRegex.Match(text);
+            if (fromMatch.Success)
+            {
+                index = fromMatch.Index;
+            }
+            return fromMatch.Success;
+        }
+
+        public bool GetBetweenTokenIndex(string text, out int index)
+        {
+            index = -1;
+            var beforeMatch = BeforeRegex.Match(text);
+            if (beforeMatch.Success)
+            {
+                index = beforeMatch.Index;
+            }
+            return beforeMatch.Success;
+        }
+
+        public bool HasConnectorToken(string text)
+        {

```

Comment:
@yiylin, yes, your right. @rubio41, Before making a PR it's always advisable to rebase onto the latest version in the main branch.